### PR TITLE
replaced some ZLanguage calls

### PR DESCRIPTION
**Pull request by Guite**
| Q                 | A
| ----------------- | ---
| Bug fix?          | yes
| New feature?      | no
| BC breaks?        | no
| Deprecations?     | no
| Fixed tickets     | -
| Refs tickets      | -
| License           | MIT
| Changelog updated | no

### CODE REVIEW

File: src/system/CategoriesModule/Controller/ConfigController.php [hunk 1]
```diff
@@ -50,7 +50,8 @@ public function configAction(Request $request)
 
         $form = $this->createForm('Zikula\CategoriesModule\Form\Type\ConfigType',
             $modVars, [
-                'translator' => $this->get('translator.default')
+                'translator' => $this->get('translator.default'),
+                'request' => $request
```
Comment (craigh) [inline src/system/CategoriesModule/Controller/ConfigController.php]: if the only thing that is needed is the locale from the request then only send the locale

File: src/system/CategoriesModule/Form/Type/ConfigType.php [hunk 2]
```diff
@@ -121,7 +122,7 @@ private function getCategoryChoices()
         $category = CategoryUtil::getCategoryByID(1);
         $categoryList = CategoryUtil::getSubCategoriesForCategory($category, $recurse, $relative, $includeRoot, $includeLeaf, $all, '', '', null, 'sort_value');
 
-        $lang = ZLanguage::getLanguageCode();
+        $lang = !empty($request) ? $request->getLocale() : 'en';
```
Comment (craigh) [inline src/system/CategoriesModule/Form/Type/ConfigType.php]: in this Form class, you need to set a default value for the arg (in `setDefaults()`) so you can set the `en` value there.

File: src/system/CategoriesModule/Twig/Extension/CategoriesExtension.php [hunk 3]
```diff
@@ -13,13 +13,24 @@
...
+
+    /**
+     * CategoriesExtension constructor.
+     *
+     * @param RequestStack $requestStack
+     */
+    public function __construct(RequestStack $requestStack)
     {
-        return 'zikula_categories_module';
+        $this->request = $requestStack->getMasterRequest();
```
Comment (craigh) [inline src/system/CategoriesModule/Twig/Extension/CategoriesExtension.php]: good practice is to delay getting the request from the stack until you need the value.

File: src/system/ExtensionsModule/Api/VariableApi.php [hunk 4]
```diff
@@ -27,7 +27,7 @@ class VariableApi
 {
     const CONFIG = 'ZConfig';
 
-    private $isInitialized = false;
+    private $isInitialised = false;
```
Comment (craigh) [inline src/system/ExtensionsModule/Api/VariableApi.php]: please don't change this. I wrote the class, so I get to use my version of English.

File: src/system/ExtensionsModule/Api/VariableApi.php [hunk 5]
```diff
@@ -86,11 +87,12 @@ public function __construct(
      *
      * @return void
      */
-    private function initialize()
+    private function initialise()
```
Comment (craigh) [inline src/system/ExtensionsModule/Api/VariableApi.php]: same comment as above wrt English spelling.

**Comment by Guite** [comment]
done

**Comment by craigh** [comment]
👍

File: src/system/CategoriesModule/Twig/Extension/CategoriesExtension.php [hunk 6]
```diff
@@ -163,7 +163,7 @@ public function moduleTableSelector($modname = null, $name = null, $selectedValu
     public function categorySelector($category = 0, $field = 'id', $name = null, $selectedValue = 0, $defaultValue = 0, $defaultText = '',
         $recurse = true, $relative = true, $includeRoot = false, $includeLeaf = true)
     {
-        $lang = !empty($this->requestStack->getMasterRequest()) ? $this->requestStack->getMasterRequest()->getLocale() : 'en';
+        $lang = $this->requestStack->getMasterRequest() != '' ? $this->requestStack->getMasterRequest()->getLocale() : 'en';
```
Comment (craigh) [inline src/system/CategoriesModule/Twig/Extension/CategoriesExtension.php]: is there a `hasRequest` method?

Comment (Guite) [inline src/system/CategoriesModule/Twig/Extension/CategoriesExtension.php]: no

**Pull request merged**